packages/core/parcel-bundler/src/scope-hoisting/concat.js

```diff
@@ -161,6 +163,29 @@ module.exports = (packager, ast) => {
             if (!isUnusedValue(path)) {
               let name = getName(mod, 'exports');
               node = t.identifier(replacements.get(name) || name);
+
+              // Insert __esModule interop flag if the required module is an ES6 module with a default export.
+              // This ensures that code generated by Babel and other tools works properly.
+              if (
+                asset.cacheData.isCommonJS &&
```

Comment:
@mischnic was this ported to v2?

Reply:
Yes (actually, Devon ported these to Parcel 1): https://github.com/parcel-bundler/parcel/blob/cb18d4507647f31824daeab1dc9de9428321a02a/packages/shared/scope-hoisting/src/link.js#L190-L215